src/main/java/org/terasology/metalrenegades/ai/component/AgentComponent.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.metalrenegades.ai.component;
+
+import org.terasology.entitySystem.Component;
+
+/**
+ * A component added to prefabs containing spawnable AI agents.
+ */
+public class AgentComponent implements Component {
```

Comment:
Maybe use a more specific term here than `Agent` which might be at high risk for colliding with actual AI architectural systems? How about `Resident` ?

Reply:
The agent component and references to 'agent' have been changed to "resident" as needed.